bin/p2-exec/main.go

```diff
@@ -214,31 +250,22 @@ func loadEnvDir(dir string) error {
 // a cgroup with the name <cgroupname> will be created, using the parameters for
 // <launchablename> found in the platform configuration
 // then, the current PID will be added to that cgroup
-func cgEnter(platconf, launchableName, cgroupName string) error {
-	platconfBuf, err := ioutil.ReadFile(platconf)
-	if err != nil {
-		return err
-	}
-	cgMap := make(map[string]map[string]cgroups.Config)
-	err = yaml.Unmarshal(platconfBuf, cgMap)
-	if err != nil {
-		return err
-	}
-
-	if _, ok := cgMap[launchableName]; !ok {
-		return util.Errorf("Unknown launchable %q in PLATFORM_CONFIG_PATH", launchableName)
-	}
-	if _, ok := cgMap[launchableName]["cgroup"]; !ok {
-		return util.Errorf("Launchable %q has malformed PLATFORM_CONFIG_PATH", launchableName)
+func cgCreate(cgroupName string, targetDirectory string, config cgroups.Config) error {
```

Comment:
I like what you did here, it's much easier to follow what this function does now